tests/functional/object/rgw/test_rgw_sts.py

```
@@ -524,3 +520,306 @@ def test_rgw_sts_configuration(
+            assert False, "Expected error with invalid Session Token but got success"
+        except ClientError as e:
+            error_code = e.response["Error"]["Code"]
+            logger.info(
+                f"Step 7: Got expected error with invalid Session Token - Error code: {error_code}"
+            )
+            assert (
+                error_code == "InvalidArgument"
+            ), f"Unexpected error code for invalid session token: {error_code}"
+
+        # Step 8: Wait for STS token to expire
+        logger.info(f"Step 8: Waiting {token_duration + 10}s for STS token to expire")
```

Ruff (0.15.15)

[warning] Logging statement uses f-string

(G004)